### PR TITLE
Adding a #user-operation message for each managed Data Source created…

### PR DESCRIPTION
**Pull request by lasryaric**
# Description

Adding a #user-operation message for each managed Data Source created or updated.
The goal is to add visibility to the life of managed data sources as we often have issue where our users modified something on the connector side and we spend a lot of time tracking that down.

This is a "best effort" posting to Slack, as we don't want to rely on the Slack API for the creation of all managed Data Sources.

Screenshot 2024-02-01 at 13 09 09

## Deploy Plan

Set the two var envs to front and deploy front:
```
SLACK_USER_OPERATION_BOT_TOKEN
SLACK_USER_OPERATION_CHANNEL_ID
```

### CODE REVIEW

File: front/pages/api/w/[wId]/data_sources/[name]/managed/update.ts [hunk 1]
```diff
@@ -104,6 +104,15 @@ async function handler(
         },
       });
 
+      void sendUserOperationMessage({
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/data_sources/[name]/managed/update.ts:107]: Can we log only if the `lastEditedById` changes?

File: types/src/shared/user_operation.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  logger,
+}: {
+  message: string;
+  logger: LoggerInterface;
+}) {
+  const { SLACK_USER_OPERATION_BOT_TOKEN, SLACK_USER_OPERATION_CHANNEL_ID } =
+    process.env;
+
+  if (!SLACK_USER_OPERATION_BOT_TOKEN || !SLACK_USER_OPERATION_CHANNEL_ID) {
+    console.log(
```
Comment (flvndvd) [inline types/src/shared/user_operation.ts:14]: Let's use `logger`, no?

File: front/pages/api/w/[wId]/data_sources/[name]/managed/update.ts (hunk 1 above)
Comment (lasryaric) [reply front/pages/api/w/[wId]/data_sources/[name]/managed/update.ts:107]: The point is to log when the user went through the OAuth flow, not only when the editedby changed.

Comment (flvndvd) [reply front/pages/api/w/[wId]/data_sources/[name]/managed/update.ts:107]: I don't see a problem with users undergoing the OAuth process multiple times. The key concern, in my opinion, is when a different person initiates the OAuth flow from what we've identified. I just want to ensure that those logs will be actionable.

**Review by flvndvd — APPROVED** [review]
No strong opinion, but left a comment.

**Pull request merged**